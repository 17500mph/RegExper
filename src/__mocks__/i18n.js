const i18n = jest.requireActual('i18n');

// Load empty resource bundle to reduce logging output
i18n.default.addResourceBundle('dev', 'translation', {});
i18n.default.addResourceBundle('en', 'translation', {});
i18n.default.addResourceBundle('other', 'translation', {});

module.exports = {
  ...i18n,
  locales: [
    { code: 'en', name: 'English' },
    { code: 'other', name: 'Other' }
  ],
  mockT: str => `TRANSLATE(${ str })`
};
