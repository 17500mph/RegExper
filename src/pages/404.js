import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation, Trans } from 'react-i18next';

import Metadata from 'components/Metadata';
import Message from 'components/Message';

export const ErrorPage = ({ t }) => <>
  <Metadata title={ t('Page Not Found') } />
  <Message type="error" heading={ t('404 Page Not Found') }>
    <p><Trans>The page you have requested could not be found.</Trans></p>
  </Message>
</>;

ErrorPage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(ErrorPage);
