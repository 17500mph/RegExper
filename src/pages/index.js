import React from 'react';
import PropTypes from 'prop-types';
import { Link, graphql } from 'gatsby';
import URLSearchParams from '@ungap/url-search-params';

import Metadata from 'components/Metadata';
import Message from 'components/Message';
import App from 'components/App';

export const query = graphql`
  query IndexPageQuery {
    site {
      siteMetadata {
        defaultSyntax
        syntaxList { id, label }
      }
    }
  }
`;

const readURLHash = (location, defaultSyntax) => {
  const query = location.hash.slice(1);
  const params = new URLSearchParams(query);
  const permalinkUrl = query ? location.href : null;

  if (params.get('syntax')) {
    return {
      syntax: params.get('syntax'),
      expr: params.get('expr'),
      permalinkUrl
    };
  } else {
    // Assuming old-style URL
    return {
      syntax: defaultSyntax,
      expr: query,
      permalinkUrl
    };
  }
};

export const IndexPage = ({
  location,
  data: { site: { siteMetadata } }
}) => <>
  <Metadata/>
  <noscript>
    <Message type="error" heading="JavaScript Required">
      <p>You need JavaScript to use Regexper.</p>
      <p>If you have concerns regarding the use of tracking code on Regexper,
        please see the <Link to="/privacy">Privacy Policy</Link>.</p>
    </Message>
  </noscript>
  <App
    syntaxList={ siteMetadata.syntaxList }
    { ...readURLHash(location, siteMetadata.defaultSyntax) } />
</>;

IndexPage.propTypes = {
  location: PropTypes.object,
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        defaultSyntax: PropTypes.string,
        syntaxList: PropTypes.arrayOf(PropTypes.shape({
          id: PropTypes.string,
          label: PropTypes.string
        }))
      })
    })
  })
};

export default IndexPage;
