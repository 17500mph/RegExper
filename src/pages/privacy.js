import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';

import Metadata from 'components/Metadata';
import PrivacyPolicy from 'components/PrivacyPolicy';

export const PrivacyPage = ({ t }) => <>
  <Metadata title={ t('Privacy Policy') } />
  <PrivacyPolicy />
</>;

PrivacyPage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(PrivacyPage);
