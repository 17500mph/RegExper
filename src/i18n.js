import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-xhr-backend';

export const locales = [
  { code: 'en', name: 'English' },
  { code: 'en-AC', name: 'English (ALL CAPS)' }
];

i18n
  .use(LanguageDetector)
  .use(Backend)
  .init({
    fallbackLng: 'en',
    debug: false,
    keySeparator: null,
    nsSeparator: null,
    saveMissing: true,
    saveMissingTo: 'current',
    missingKeyHandler: (lng, ns, key, fallback) => {
      if (!i18n.getResourceBundle(lng)) {
        return; // Don't bother logging if the resource bundle isn't loaded
      }

      const escapedKey = key.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
      // eslint-disable-next-line no-console
      if (console && console.group && console.log) {
        // eslint-disable-next-line no-console
        console.group(`Missing key in ${ lng }`);
        // eslint-disable-next-line no-console
        console.log(`"${ escapedKey }": |\n  ${ fallback }`);
        // eslint-disable-next-line no-console
        console.groupEnd();
      }
    },
    interpolation: {
      escapeValue: false
    },
    backend: {
      loadPath: '{{lng}}',
      parse: data => data,
      ajax: async (lng, options, callback) => {
        try {
          const { default: locale } = await import(`locales/${ lng }.yaml`);

          callback(locale, { status: '200' });
        }
        catch (e) {
          callback(null, { status: '500' });
        }
      }
    }
  });

export default i18n;
