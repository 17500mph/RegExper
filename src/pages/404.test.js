import React from 'react';
import { shallow } from 'enzyme';

import { mockT } from 'i18n';
import { ErrorPageImpl } from 'pages/404';

describe('Error Page', () => {
  test('rendering', () => {
    const component = shallow(
      <ErrorPageImpl t={ mockT } />
    );
    expect(component).toMatchSnapshot();
  });
});
