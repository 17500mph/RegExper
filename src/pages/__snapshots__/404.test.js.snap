// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page rendering 1`] = `
<Fragment>
  <withI18nextTranslation(Metadata)
    title="TRANSLATE(Page Not Found)"
  />
  <Message
    heading="TRANSLATE(404 Page Not Found)"
    type="error"
  >
    <p>
      <Trans>
        The page you have requested could not be found.
      </Trans>
    </p>
  </Message>
</Fragment>
`;
